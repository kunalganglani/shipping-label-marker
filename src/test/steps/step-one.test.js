import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import GetSenderAddress from '../../features/shipping-label-maker/steps/step-one';

Enzyme.configure({ adapter: new Adapter() });
const wizardContext = {
  from: {
    name: '',
    street: '',
    city: '',
    state: '',
    zip: '',
  },
  to: {
    name: '',
    street: '',
    city: '',
    state: '',
    zip: '',
  },
  weight: {
    weightValue: '',
    weightUnit: 'Pounds',
  },
  shippingOption: '1',
};

test('GetSenderAddress renders correctly', () => {
  const handleAction = jest.fn();
  const tree = renderer
    .create(<GetSenderAddress wizardContext={wizardContext.from} onAction={handleAction} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('createAction is called', () => {
  const someFunction = jest.fn();
  const wrapper = shallow(<GetSenderAddress wizardContext={wizardContext.from} onAction={someFunction} />);
  wrapper.find('WithStyles(ForwardRef(TextField))').first().simulate('change', {
    target: {
      value: 1,
    },
  });
  expect(someFunction).toHaveBeenCalled();
});
