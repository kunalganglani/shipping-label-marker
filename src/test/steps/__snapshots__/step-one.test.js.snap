// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSenderAddress renders correctly 1`] = `
Array [
  <h3>
    Enter the sender's address
  </h3>,
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="name"
      id="name-label"
    >
      Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="street"
      id="street-label"
    >
      Street
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="street"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Street
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <div
    className="css-14dkja1"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="city"
        id="city-label"
      >
        City
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="city"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              City
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="state"
        id="state-label"
      >
        State
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="state"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              State
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="zip"
        id="zip-label"
      >
        Zip
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="zip"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Zip
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
]
`;
